event-based-simulators/extras/ImportData.py

```diff
@@ -0,0 +1,124 @@
+
+log_format = '[%(asctime)s] [%(levelname)s] - %(message)s'
+log = logging.getLogger('ImportData')
+c8y = ArgumentsAndCredentialsHandler.c8yPlatformConnection()
+format = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+C8Y_PROFILE_GROUP = 'c8y_EventBasedSimulatorProfile'
+
+user_and_pass_bytes = base64.b64encode(
+    (Environment.C8Y_TENANT + "/" + Environment.C8Y_USER + ':' + Environment.C8Y_PASSWORD).encode('ascii'))  # bytes
+user_and_pass = user_and_pass_bytes.decode('ascii')  # decode to str
+
+C8Y_HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Authorization': 'Basic ' + user_and_pass
+}
+
+MEASUREMENTS_HEADERS = {
+    'Content-Type': 'application/vnd.com.nsn.cumulocity.measurementcollection+json',
+    'Accept': 'application/json',
+    'Authorization': 'Basic ' + user_and_pass
+}
+
+def getDeviceIdByExternalId(external_id):
+    log.info(f'Searching for device with ext ID {external_id}')
+    response = requests.get(f'{Environment.C8Y_BASE}/identity/externalIds/{C8Y_PROFILE_GROUP}/{external_id}', headers=C8Y_HEADERS)
+    if response.ok:
+        device_id = response.json()['managedObject']['id']
+        log.info(f'Device({device_id}) has been found by its external id "{C8Y_PROFILE_GROUP}/{external_id}".')
+        return device_id
+    log.warning(f'No device has been found for the external id "{C8Y_PROFILE_GROUP}/{external_id}".')
+    return None
+
+def createAlarm(alarm):
+    response = requests.post(f'{Environment.C8Y_BASE}/alarm/alarms', headers=C8Y_HEADERS, data=json.dumps(alarm))
+    if response.ok:
+        return response.json()
+    log.warning(response)
+    return None
+
+def createMeasurements(measurements):
+    response = requests.post(f'{Environment.C8Y_BASE}/measurement/measurements', headers=MEASUREMENTS_HEADERS, data=json.dumps(measurements))
+    if response.ok:
+        return response.json()
+    log.warning(response)
+    return None
+
+def getTimeDifference(object, key):
+    creation_Time = datetime.strptime(object[key], format)
+    now = datetime.utcnow()
+    return (now - creation_Time)
+
+def deleteUnwantedFields(alarm):
+    del alarm['lastUpdated']
+    del alarm['count']
+    del alarm['creationTime']
+    del alarm['history']
+    return alarm
+
+def importAlarms(alarms, id):
+    log.debug('Importing all alarms')
+    log.debug(f'Alarms:{alarms}')
+    timeShift = getTimeDifference(alarms[0], 'creationTime')
+    for alarm in alarms:
+        alarm['source']['id'] = id
+        alarm = deleteUnwantedFields(alarm)
+        alarm['time'] = (datetime.strptime(alarm['time'], format) + timeShift).strftime(format)
+        log.debug(f'Posting Alarm for device {id}: {alarm}')
+        createAlarm(alarm)
+
+def importMeasurements(measurements, id):
+    log.debug('Importing all measurements')
+    log.debug(f'Measurements: {measurements}')
+    timeShift = getTimeDifference(measurements[len(measurements)-1], 'time')
+    for i in range(len(measurements)):
```

Comment:
is there a reason why you dont use `for measurment in measurements`?

Reply:
Yes since it would take the object from the array and when we would change id... in that object it would not be changed in the object of the array, but instead in the copy of the object. And since we are pushing the whole array after the loop instead of every single one, I figured it might be best to do it like this

Reply:
ah, ok. so the difference to the loop in importAlarms() is that each alarm is created individually.